server/src/main/java/org/apache/druid/segment/loading/StorageLocationSelectorStrategy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.loading;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.druid.timeline.DataSegment;
+
+import java.util.Iterator;
+
+/**
+ * This interface describes the storage location selection strategy which is responsible for ordering the
+ * available multiple {@link StorageLocation}s for segment distribution.
+ *
+ * Only a snapshot of the locations is returned here. The implemntations currently do not handle all kinds of
+ * concurrency issues and accesses to the underlying storage. Please see
+ * https://github.com/apache/incubator-druid/pull/8038#discussion_r325520829 of PR https://github
+ * .com/apache/incubator-druid/pull/8038 for more details.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "tier", defaultImpl = LeastBytesUsedStorageLocationSelectorStrategy.class)
```

Comment:
Is `property = "tier"` here required, or is it copy/pasted from another location (like `TierSelectorStrategy`)?

Reply:
good catch, should probably be "type"

Reply:
@dclim, let me know if it needs to be removed or changed to `type`.

Reply:
Changed it to type.